crypto/fipsmodule/rand/urandom_test.cc

```diff
@@ -257,6 +268,32 @@ static void GetTrace(std::vector<Event> *out_trace, unsigned flags,
         break;
       }
 
+      case __NR_nanosleep: {
+        // First true condition: In FIPS mode, an |ioctl| call with command
+        // |RNDGETENTCNT| is used. If this fails, a delay is injected. The
+        // failure happens when the test flag |URANDOM_NOT_READY| is set. But
+        // since this bit is cleared below we detect this event using
+        // |urandom_not_ready_was_cleared|.
+        //
+        // Second true condition: We can have two or more consecutive
+        // |nanosleep| calls. This happens if |nanosleep| returns -1. The PRNG
+        // model only accounts for one |nanosleep| call. Do the same here.
```

Comment:
I don't understand this part; so will nanosleep be only called once or more, based on the loop I commented on in urandom.c, l.145, 433?

Reply:
If `nanosleep` fails (returns `-1`), there is an additional retry logic that will cause this code to catch more `nanosleep` calls.

This is not accounted for in the PRNG model, because there is no way we can predict the number of (potential, rare(?)) retries.

What we do know is that there should always be one `nanosleep` call (in the test dimensions testing failures). I ignore any consecutive `nanosleep` calls. Without this, this test is unreliable and can fail at random .

Reply:
Thank you for the clarification. Just for my understanding, "Without this" means "without ignoring consecutive `nanosleep` calls" or "without accounting for this case in the checks below"?

Reply:
Must ignore consecutive calls because these are not included in the PRNG model.